piptools/resolver.py

```diff
@@ -85,9 +84,6 @@ def resolve(self, max_rounds=10):
             self.dependency_cache.clear()
             self.repository.clear_caches()
 
-        self.check_constraints(chain(self.our_constraints,
```

Comment:
We still have a limitation on constraints: We don't support file link (as checked in `is_url_requirement`), so we still need to make a check to warn the user, both here and in `pip-sync`.